plugins/modules/ndo_pod_profile.py

```diff
@@ -0,0 +1,357 @@
+    mso_template.validate_template("fabricResource")
+
+    match = mso_template.get_pod_profile_object(uuid, name)
+
+    if (uuid or name) and match:
+        set_pod_profile_policy_names(mso, mso_templates, match.details)
+        mso.existing = mso.previous = copy.deepcopy(match.details)  # Query a specific object
+    elif match:
+        mso.existing = [set_pod_profile_policy_names(mso, mso_templates, obj) for obj in match]  # Query all objects
+
+    pod_profile_path = "/fabricResourceTemplate/template/podProfiles/{0}".format(match.index if match else "-")
+
+    ops = []
+
+    if state == "present":
+
+        mso_values = {
+            "name": name,
+            "description": description,
+        }
+
+        if pods:
+            mso_values["blocks"] = pods
+            mso_values["kind"] = "podRange"
+        elif pods == []:
+            mso_values["kind"] = "all"
+
+        if pod_settings_uuid:
+            mso_values["policy"] = pod_settings_uuid
+        else:
+            if not pod_settings_template and not pod_settings_template_id:
+                module.fail_json(msg="pod_settings is set but any of the following are missing: pod_settings_template, pod_settings_template_id")
+            pod_settings_mso_template = mso_templates.get_template("fabric_policy", pod_settings_template, pod_settings_template_id)
+            mso_values["policy"] = pod_settings_mso_template.get_pod_settings_object(pod_settings_uuid, pod_settings, fail_module=True).details.get("uuid")
+
+        if match:
+            remove_data = []
+            if pods == []:
+                remove_data.append("blocks")
+
+            append_update_ops_data(ops, match.details, pod_profile_path, mso_values, remove_data)
+            mso.sanitize(mso_values, collate=True, unwanted=remove_data)
+            set_pod_profile_policy_names(mso, mso_templates, mso.proposed)
+        else:
+            if not pods:
+                mso_values["kind"] = "all"
+            mso.sanitize(mso_values)
+            ops.append(dict(op="add", path=pod_profile_path, value=mso.sent))
+
+    elif state == "absent":
+        if match:
+            ops.append(dict(op="remove", path=pod_profile_path))
+
+    if not module.check_mode and ops:
+        response = mso.request(mso_template.template_path, method="PATCH", data=ops)
+        match = mso_template.get_pod_profile_object(uuid, name, search_object=response)
+        if match:
+            set_pod_profile_policy_names(mso, mso_templates, match.details)
+            mso.existing = match.details  # When the state is present
+        else:
+            mso.existing = {}  # When the state is absent
+    elif module.check_mode and state != "query":  # When the state is present/absent with check mode
+        set_pod_profile_policy_names(mso, mso_templates, mso.proposed)
+        mso.existing = mso.proposed if state == "present" else {}
+    mso.exit_json()
+
+
+def set_pod_profile_policy_names(mso, mso_templates, pod_profile):
```

Comment:
Should this function also be adding the templateName/templateId of the pod profile?

Reply:
will add